src/sentry/api/endpoints/project_rules_configuration.py

```diff
@@ -64,7 +65,11 @@ def get(self, request: Request, project) -> Response:
             if rule_type.startswith("condition/"):
                 condition_list.append(context)
             elif rule_type.startswith("filter/"):
-                filter_list.append(context)
+                if (
+                    has_issue_severity_alerts
+                    or context["id"] != "sentry.rules.filters.issue_severity.IssueSeverityFilter"
+                ):
+                    filter_list.append(context)
```

Comment:
Echoing Steve's earlier comment, the `context` condition might be a bit easier to grok in the affirmative:

```suggestion
                if (
                    context["id"] == "sentry.rules.filters.issue_severity.IssueSeverityFilter"
                    and not has_issue_severity_alerts
                ):
                    continue
                filter_list.append(context)
```